### ISSUE TITLE
Error with v.1.3.4

### ISSUE THREAD

**Issue by nickcodefresh** [gavlyukovskiy/spring-boot-data-source-decorator#20]
I'm trying to use v1.3.4 and I'm seeing the following error:

```
org.springframework.batch.item.ItemStreamException: Error while closing item reader
	at org.springframework.batch.item.support.AbstractItemCountingItemStreamItemReader.close(AbstractItemCountingItemStreamItemReader.java:136)
	at org.springframework.batch.item.support.AbstractItemCountingItemStreamItemReader$$FastClassBySpringCGLIB$$ebb633d0.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:738)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.aop.support.DelegatingIntroductionInterceptor.doProceed(DelegatingIntroductionInterceptor.java:133)
	at org.springframework.aop.support.DelegatingIntroductionInterceptor.invoke(DelegatingIntroductionInterceptor.java:121)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:673)
	at org.springframework.batch.item.database.HibernatePagingItemReader$$EnhancerBySpringCGLIB$$771cae9e.close(<generated>)
	at org.springframework.batch.item.support.CompositeItemStream.close(CompositeItemStream.java:85)
	at org.springframework.batch.core.step.item.ChunkMonitor.close(ChunkMonitor.java:106)
	at org.springframework.batch.item.support.CompositeItemStream.close(CompositeItemStream.java:85)
	at org.springframework.batch.core.step.tasklet.TaskletStep.close(TaskletStep.java:305)
	at org.springframework.batch.core.step.AbstractStep.execute(AbstractStep.java:271)
	at org.springframework.batch.core.job.SimpleStepHandler.handleStep(SimpleStepHandler.java:148)
	at org.springframework.batch.core.job.AbstractJob.handleStep(AbstractJob.java:392)
	at org.springframework.batch.core.job.SimpleJob.doExecute(SimpleJob.java:135)
	at org.springframework.batch.core.job.AbstractJob.execute(AbstractJob.java:306)
	at org.springframework.batch.core.launch.support.SimpleJobLauncher$1.run(SimpleJobLauncher.java:135)
	at org.springframework.core.task.SimpleAsyncTaskExecutor$ConcurrencyThrottlingRunnable.run(SimpleAsyncTaskExecutor.java:271)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException: null
	at com.github.gavlyukovskiy.cloud.sleuth.TracingQueryExecutionListener.afterMethod(TracingQueryExecutionListener.java:131)
	at net.ttddyy.dsproxy.listener.CompositeMethodListener.afterMethod(CompositeMethodListener.java:25)
	at net.ttddyy.dsproxy.listener.MethodExecutionListenerUtils.invoke(MethodExecutionListenerUtils.java:53)
	at net.ttddyy.dsproxy.proxy.ConnectionProxyLogic.invoke(ConnectionProxyLogic.java:47)
	at net.ttddyy.dsproxy.proxy.jdk.ConnectionInvocationHandler.invoke(ConnectionInvocationHandler.java:25)
	at com.sun.proxy.$Proxy161.close(Unknown Source)
	at org.hibernate.engine.jdbc.connections.internal.DatasourceConnectionProviderImpl.closeConnection(DatasourceConnectionProviderImpl.java:127)
	at org.hibernate.internal.AbstractSessionImpl$NonContextualJdbcConnectionAccess.releaseConnection(AbstractSessionImpl.java:397)
	at org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl.releaseConnection(LogicalConnectionManagedImpl.java:172)
	at org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl.close(LogicalConnectionManagedImpl.java:215)
	at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.close(JdbcCoordinatorImpl.java:193)
	at org.hibernate.internal.SessionImpl.close(SessionImpl.java:421)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:216)
	at org.springframework.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:201)
	at org.springframework.batch.item.database.HibernateItemReaderHelper.close(HibernateItemReaderHelper.java:204)
	at org.springframework.batch.item.database.HibernatePagingItemReader.doClose(HibernatePagingItemReader.java:178)
	at org.springframework.batch.item.support.AbstractItemCountingItemStreamItemReader.close(AbstractItemCountingItemStreamItemReader.java:133)
	... 21 common frames omitted

```

**Comment by gavlyukovskiy** [comment]
I think I fixed similar bug with p6spy implementation - https://github.com/gavlyukovskiy/spring-boot-data-source-decorator/issues/18, but forgot to do the same for datasource-proxy. 
I assume that you're not closing resources in reverse order (ResultSet -> Statement -> Connection), this is what I fixed for p6spy. Do you have any code sample to verify this? Also if you can try to reproduce issue with p6spy (generates same traces as datasource-proxy) it would be helpful,

**Comment by gavlyukovskiy** [comment]
Fixed in https://github.com/gavlyukovskiy/spring-boot-data-source-decorator/commit/52ab82eb8b6e094f0f27afa5fe4ff4034285b478, published under release 1.3.5.
